src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -728,14 +746,21 @@ private void Resize(int newSize, bool forceNewHashCodes)
 
             if (!typeof(TKey).IsValueType && forceNewHashCodes)
             {
+                Debug.Assert(_comparer is INonRandomizedEqualityComparer<TKey>);
+                _comparer = ((INonRandomizedEqualityComparer<TKey>)_comparer).GetRandomizedComparer();
```

Comment:
Doesn't this mean that we'll be serializing the randomized comparer instance after a resize? Is this desirable?

Reply:
The non-randomized comparer is strictly an implementation detail of the dictionary. It is never serialized out. On the off-chance that it _is_ serialized out, it lies about its type so that the serialized data blob contains a reference back to the underlying randomized comparer.

https://github.com/dotnet/runtime/blob/5f2398b1fd864fd988027e35b5cab2b128242720/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/NonRandomizedStringEqualityComparer.cs#L28-L32

So this line shouldn't be a breaking behavioral change.